flax/linen/transforms.py

```diff
@@ -1262,3 +1262,40 @@ def core_fn(scopes, *args, **kwargs):
       return trafo_fn(module_scopes, *args, **kwargs)
 
   return wrapped_fn
+
+def switch(index, branches: Sequence[Module], *args, **kwargs):
+  """Lifted version of `jax.lax.switch`."""
+
+  all_variables = [module.scope.variables() for module in branches]
```

Comment:
Switch is esentially a generalization of cond, right? Probably the code between nn.cond and nn.switch can be mostly shared. This implementation doesn't actually uses the lifting logic so it's missing all the weird edge cases it handles